core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -4195,6 +4195,24 @@ public void doRestart(StaplerRequest req, StaplerResponse rsp) throws IOExceptio
             rsp.sendRedirect2(".");
         }
     }
+    
+    /**
+     * REST client version of {@link #doRestart(StaplerRequest, StaplerResponse)}
+     * Perform a restart of Jenkins, if we can.
+     *
+     * This first replaces "app" to {@link HudsonIsRestarting}
+     *
+     * @since TODO
+     */
+    @RequirePOST
+    @Restricted(NoExternalUse.class)
+    public void doRestartSimple(StaplerRequest req, StaplerResponse rsp) throws RestartNotSupportedException {
```

Comment:
I'm not a fan of this URL name. Is there no way to reuse the same /restart endpoint here?

Reply:
Due to all the discussions until now, there is no proposal made by reusing the endpoint that was satisfying.

Reply:
Well, I suppose we can always make the `/restart` API support both styles later on when we have more sophisticated web framework features to easily support it.

Reply:
Oh, now that I look more closely, I can see why. This URL name is about as good as it gets without making the HTTP headers more complicated elsewhere.